subsys/bluetooth/audio/csip_set_member.c

```diff
@@ -492,8 +492,11 @@ static void csip_security_changed(struct bt_conn *conn, bt_security_t level,
 static void handle_csip_disconnect(struct bt_csip_set_member_svc_inst *svc_inst,
 				   struct bt_conn *conn)
 {
-	LOG_DBG("Non-bonded device");
 	if (is_last_client_to_write(svc_inst, conn)) {
+		if (bt_addr_le_is_bonded(conn->id, &conn->le.dst)) {
+			return;
+		}
+		LOG_DBG("Non-bonded device");
```

Comment:
What about the `for` loop in line 515 below? Shouldn't it be:
`if (is_last_client_to_write(svc_inst, conn) && !bt_addr_le_is_bonded(conn->id, &conn->le.dst))` ?

Reply:
No - but we should also do the cleanup for a bonded connection!
Will fix.

Reply:
You were right.  We need to extend the check to the for loop, when cleaning up.  I have included that fix.